### DISCUSSION TITLE
Deployment failed

### DISCUSSION THREAD

**Discussion by Ynsinan** [vercel/vercel#5927] [Help]
21:00:09 | Cloning github.com/Ynsinan/quoteapp (Branch: master, Commit: f6fe7c7)
-- | --
21:00:10 | Cloning completed in 558ms
21:00:10 | Analyzing source code...
21:00:10 | Installing build runtime...
21:00:13 | Build runtime installed: 2878.660ms
21:00:15 | Looking up build cache...
21:00:15 | Build cache not found
21:00:16 | Detected package.json
21:00:16 | Installing dependencies...
21:00:41 | > core-js@2.6.12 postinstall /vercel/workpath0/node_modules/babel-runtime/node_modules/core-js
21:00:41 | > node -e "try{require('./postinstall')}catch(e){}"
21:00:41 | > core-js@3.9.1 postinstall /vercel/workpath0/node_modules/core-js
21:00:41 | > node -e "try{require('./postinstall')}catch(e){}"
21:00:41 | > core-js-pure@3.9.1 postinstall /vercel/workpath0/node_modules/core-js-pure
21:00:41 | > node -e "try{require('./postinstall')}catch(e){}"
21:00:41 | > ejs@2.7.4 postinstall /vercel/workpath0/node_modules/ejs
21:00:41 | > node ./postinstall.js
21:00:43 | npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules/webpack-dev-server/node_modules/fsevents):
21:00:43 | npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
21:00:43 | npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules/watchpack-chokidar2/node_modules/fsevents):
21:00:43 | npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
21:00:43 | npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.3.2 (node_modules/fsevents):
21:00:43 | npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
21:00:43 | added 1944 packages from 758 contributors in 27.194s
21:00:44 | 132 packages are looking for funding
21:00:44 | run `npm fund` for details
21:00:45 | Running "npm run build"
21:00:45 | > quoteapp@0.1.0 build /vercel/workpath0
21:00:45 | > react-scripts build
21:00:46 | Creating an optimized production build...
21:00:53 | Treating warnings as errors because process.env.CI = true.
21:00:53 | Most CI servers set it automatically.
21:00:53 | Failed to compile.
21:00:53 | src/App.js
21:00:53 | Line 50:6:  React Hook useEffect has a missing dependency: 'getQuote'. Either include it or remove the dependency array  react-hooks/exhaustive-deps
21:00:53 | npm ERR! code ELIFECYCLE
21:00:53 | npm ERR! errno 1
21:00:53 | npm ERR! quoteapp@0.1.0 build: `react-scripts build`
21:00:53 | npm ERR! Exit status 1
21:00:53 | npm ERR!
21:00:53 | npm ERR! Failed at the quoteapp@0.1.0 build script.
21:00:53 | npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
21:00:53 | npm ERR! A complete log of this run can be found in:
21:00:53 | npm ERR!     /vercel/.npm/_logs/2021-03-04T18_00_53_737Z-debug.log
21:00:53 | Error: Command "npm run build" exited with 1
21:00:56 | Done with "package.json"

Guys help pls!!

**Comment by Ynsinan** [comment]
I'm getting this error while trying to deploy, couldn't find how to solve it

**Comment by wei** [comment]
The error clearly shows
> ```Line 50:6: React Hook useEffect has a missing dependency: 'getQuote'. Either include it or remove the dependency array react-hooks/exhaustive-deps```

Perhaps you can try fixing that?

**Reply by Ynsinan** [reply]
How can i fix? I try everything for this line

**Reply by wei** [reply]
You should read up on `react-hooks/exhaustive-deps`

I've fixed it for you at https://github.com/Ynsinan/quoteapp/pull/1

**Reply by Ynsinan** [reply]
When I do what you say, the program does not work as I want

**Reply by wei** [reply]
Fixed, you needed to `useCallback` to make sure the function is not redefined on each re-render.

**Reply by Ynsinan** [reply]
Thanks dude!!

**Reply by wei** [reply]
Feel free to mark my answer as the Answer if it's working now on your end~
